### PR TITLE
Added softlayer lookup helpers to the common package. Get datacenter …

### PR DESCRIPTION
**Pull request by tmulhern3**
…by name implemented. Unit tests completed as well

### CODE REVIEW

File: common/softlayer_lookup_helpers.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	datatypes "github.com/TheWeatherCompany/softlayer-go/data_types"
+	"github.com/TheWeatherCompany/softlayer-go/softlayer"
+)
+
+const (
+	DATACENTER_TYPE_NAME = "SoftLayer_Location_Datacenter"
+)
+
+func GetDatacenterByName(client softlayer.Client, name string) (int, error) {
+	response, errorCode, err := client.GetHttpClient().DoRawHttpRequest(fmt.Sprintf("%s/getDatacenters.json", DATACENTER_TYPE_NAME), "GET", new(bytes.Buffer))
```
Comment (renier) [inline common/softlayer_lookup_helpers.go]: Could we add an objectFilter on the location.Name?

File: common/softlayer_lookup_helpers.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		return -1, err
+	}
+
+	for _, location := range locations {
+		if location.Name == name {
+			return location.Id, nil
+		}
+	}
+
+	return -1, nil
```
Comment (renier) [inline common/softlayer_lookup_helpers.go]: Feels like here, there should be an error added with `fmt.Errorf("Datacenter %s not found", name)`

**Comment by renier** [comment]
General comment, what do you think of emulating terraform and creating a `helpers` package to put this in on softlayer-go

**Comment by tmulhern3** [comment]
@renier I agree with all three comments. Updating now

**Comment by renier** [comment]
...or just call the file `helpers.go`... seeing that the common package was already being used to put utilities in. Up to you.

**Pull request merged**